### PR TITLE
Rework encryption configuration api

### PR DESCRIPTION
**Pull request by timbussmann**
`c.RijndaelEncryptionService("1st", Keys)` becomes `c.EnableMessagePropertyEncryption(new RijndaelEncryptionService("1st", Keys))`

custom conventions can be defined as a second parameter to `EnableMessagePropertyEncryption`, e.g. `c.EnableMessagePropertyEncryption(new RijndaelEncryptionService("1st", Keys), t => t.Name.StartsWith("Encrypted"));`

Thoughts @SimonCropp @kbaley 
cc @ramonsmits @MarcinHoppe (as you're interested to be the maintainers for this repo)

### CODE REVIEW

**Comment by danielmarbach** [comment]
We have a few APIs which use the nested lambda approach. So how about:

```
c.MessagePropertyEncryption(e => {
   e.RijndaelEncryption("1st", Keys);
   e.Convention(t => t.Name.StartsWith(" "));
};

```
or

```
c.MessagePropertyEncryption(e => {
   e.Custom(new RijndaelEncryptionService("1st", Keys));
   e.Convention(t => t.Name.StartsWith(" "));
};
```

**Comment by timbussmann** [comment]
thought about going full lambda approach but I think it's not as nice as you would have configurations which don't work together (e.g. `RijndaelEncryption` and `Custom` [or any other encryption algorithm in case we want to provide another one). I think the approach with accepting an encryption service is simpler. For the convention alone I didn't want to create a fill Configuration-element for the lambda parameter.

**Comment by danielmarbach** [comment]
Hu? We can certainly build such an API which excludes configuration that don't work together right?

File: src/MessageProperty/IsEncrytedPropertyConvention.cs [hunk 1]
```diff
@@ -6,11 +6,13 @@
     /// <summary>
     /// Message convention definitions.
     /// </summary>
-    public class Conventions
+    class IsEncrytedPropertyConvention
```
Comment (timbussmann) [inline src/MessageProperty/IsEncrytedPropertyConvention.cs]: good catch 👍

File: src/MessageProperty/MessagePropertyEncryption.cs [hunk 2]
```diff
@@ -21,12 +22,11 @@ protected override void Setup(FeatureConfigurationContext context)
                 throw new Exception("The message property encryption extension as well as NServiceBus.Core's encryption feature are enabled. Disable one of the encryption features to avoid message payload corruption.");
```
Comment (timbussmann) [inline src/MessageProperty/MessagePropertyEncryption.cs]: 👍

**Comment by timbussmann** [comment]
> Hu? We can certainly build such an API which excludes configuration that don't work together right?

proposals?

**Comment by timbussmann** [comment]
addressed your comments @kbaley

**Pull request merged**